src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/impl/MainConsumerImpl.java

```diff
@@ -86,7 +100,12 @@ public void readAndProcessRetryTopic(@Payload List<ChipsRestInterfacesSend> mess
      * @param offset The offset of {@code data}
      * @param partition The partition of {@code data}
      */
-    private void processMessage(String consumerId, ChipsRestInterfacesSend data, Long offset, Integer partition) {
+    private void processMessage(String consumerId,
+                                ChipsRestInterfacesSend data,
+                                Long offset,
+                                Integer partition,
+                                Optional<List<String>> failedMessageOpt) {
```

Comment:
I wouldn't use optional as a param - Sonar has flagged this, i think it is only supposed to be used as a return type from a function. I'll try and attach the screen shot, it should be visible in your intellij too.
![image](https://user-images.githubusercontent.com/11330756/109487875-b8ce6700-7a7c-11eb-9f9a-eb7081c9f79d.png)